### PR TITLE
[PKG-5391] Update to 2.4.2

### PR DESCRIPTION
**Pull request by ViridianMelody**
sacrebleu 2.4.2

**Destination channel:** defaults

### Links

- [PKG-5391]
- [Upstream repository](https://github.com/mjpost/sacrebleu)
- [Upstream changelog/diff]()

### Explanation of changes:

- ~~Removed obsolete `setup.py` patch.~~ This seems to fix an error with the build, at least on Windows, so I'm leaving it in place.
- Added upstream tests.
  - `test_process_to_text` appears to be flaky on all platforms.
  - API tests fail on Windows.
- Switch to Github source, since source wasn't uploaded to Pypi.
- Drop noarch

[PKG-5391]: https://anaconda.atlassian.net/browse/PKG-5391?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,32 +1,33 @@
 {% set name = "sacrebleu" %}
-{% set version = "2.0.0" %}
+{% set version = "2.4.2" %}
```
Comment (boldorider4) [inline recipe/meta.yaml:2]: `2.4.3` is out.

**Review by boldorider4 — APPROVED** [review]
LGTM but could be upgraded to `2.4.3`

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,32 +1,33 @@
...
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: 51fb69b6683f1b9999cd180143bb6b21d7841744537c9aab235cfe676550f0cf
+  url: https://github.com/mjpost/sacrebleu/archive/refs/tags/v{{ version }}.tar.gz
```
Comment (Jrice1317) [inline recipe/meta.yaml:9]: I'm curious the reason for switching the source to the github url?

Comment (Jrice1317) [reply recipe/meta.yaml:9]: I just noticed your comment in the description, but maybe updating to `2.4.3` will remove the issue as it seems to be listed [here](https://inspector.pypi.io/project/sacrebleu/)

**Review by Jrice1317 — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml [hunk 3]
```diff
@@ -35,20 +31,34 @@ requirements:
     - tabulate >=0.8.9
     - numpy >=1.17
     - colorama
-    # extras_require={'ja': ['mecab-python3==1.0.3', 'ipadic>=1.0,<2.0']},
+    - lxml
+  run_constrained:
+    - ipadic >=1.0,<2.0
+    - mecab-python3 >=1.0.9,<2.0
+    - mecab-ko >=1.0.0,<1.0.1
+    - mecab-ko-dic >=1.0,>2.0
```
Comment (Jrice1317) [inline recipe/meta.yaml:39]:
```suggestion
    - mecab-python3 >=1.0.9,<2.0.0
    - mecab-ko >=1.0.0,<=1.0.1
    - mecab-ko-dic >=1.0,<2.0
```

**Review by Jrice1317 — APPROVED** [review]

**Pull request merged**